l10n_it_ricevute_bancarie/wizard/wizard_riba_file_export.py

```diff
@@ -234,16 +235,18 @@ def act_getfile(self):
             debitor_address = line.partner_id
             debitor_street = debitor_address.street or ''
             debitor_zip = debitor_address.zip or ''
-            if debit_bank.bank_abi and debit_bank.bank_cab:
+            debit_iban = debit_bank.acc_number.replace(" ", "") \
+                if debit_bank.acc_number else None
+            #TODO: for compatibility only remove in the next release
+            if hasattr(debit_bank, "bank_abi") and debit_bank.bank_cab:
                 debit_abi = debit_bank.bank_abi
```

Comment:
`bank_abi` è un campo introdotto da `l10n_it_abicab` quindi la dipendenza non si può rimuovere

Reply:
ha usato hasattr() per fare il check

Reply:
Mi sembra contorto, come mai farlo? Solo per poter togliere la dipendenza? Che fastidio dà?

Reply:
si, non è una cosa da usare su un modulo ufficiale secondo me

Reply:
Il motivo per cui ho tolto abicab e' che le stesse informazioni si possono ricavare dall'iban e preferisco che inseriscano l'iban non abi e cab con possibili errori. Ritengo che questo comportamento possa essere standard ed il modulo abi_cab puo' andare in disubo. Poi mi so posto il problema delle installazioni in essere o di chi proprio lo vuole ... l'introspezione e' il modo piu' veloce sicuro e senza troppa duplicazione di codice per ottenere lo scopo. Perche' toglierlo? perche' e' inutile e non vi e' relazione tra iban ed abi_cab con iban che ha precedenza ... poi il file esce "sbagliato" e vai a capire che il cliente ha messe iban ed abi/cab diversi etc etc ...dal mio punto di vista possiam toglierlo e basta senza retro. Per e' e' semplice visto che e' la prima volta che vedo e mi appresto ad usare il modulo ma magari per altri la retro e' cosa gradita.

Reply:
E i moduli che ereditano e che si aspettano che quei campi esistano?
A questo punto lo si fa al passaggio di versione questo tipo di modifica.

Reply:
Limitarsi a togliere la dipendenza sicuramente no, perchè molte installazioni (quelle che usano quei campi per generare riba) smetterebbero di funzionare.

Bisognerebbe analizzare cosa succede aggiornando un DB dopo queste modifiche e valutare se necessario script di migrazione dati.

> e' innegabile che gestire due dati (iban ed abi/cab) identici su campi diversi e' fonte di confusione

Capisco il problema alla base.
Allora piuttosto si potrebbe aggiungere un check sulla coerenza dei dati

Reply:
Suppongo che nulla accada alle installazioni in essere visto che il modulo abi_cab risulterebbe gia' installato e non verrebbe in nessun caso rimosso. L'unico problema potrebbero averlo eventuali moduli che dipendono solo da riba e che si aspettano di trovare quei fields (ma ve stanno? sian sicuri che sia corretto non mettere le depends corrette ma sperare nella catena?).

Sul discorso coerenza, oltre al gia' citato problema la wizard ragiona in maniera diversa per credit_bak e debit_bank ... sul credit usa iban sul debit prima abi/cab e successivamente iban con l'aggiunta che l'iban non poteva essere nullo altrimenti andava in errore. Non so te ma io fra qualche giorno mi saro' dimenticato di questa discussione ed al primo problema in esportazione saro' costretto a tornare a leggere il codice. Si potrebbe pensare a rendere coerenti i dati con una constraint su abi_cab mentre sul modulo riba andrebbe reso coerente il comportamento di debit e credit bank, e' una strada percorribile. Nella mia versione, purtroppo ne devo cmq fare un fork per via della dipendenza da fiscalcode, lascio cosi' com'e' con abi_cab rimosso a livello 12/OCA decidete un po' come ritenete giusto procedere. Se non rimossa la coerenza sarebbe di sicuro la strada da perseguire. Provvedo ad aprire nuova PR per il solo code clean lasciando intatte le dipendenze.

Reply:
Allora probabilmente non ho capito cosa intendevi con "Dal mio punto di vista possiam toglierlo e basta senza retro", cmq NP

> Non so te ma io fra qualche giorno mi saro' dimenticato di questa discussione ed al primo problema in esportazione saro' costretto a tornare a leggere il codice

Probabilmente questa parte potrebbe anche essere resa più chiara, anche con commenti